### PR TITLE
Update kubernetes-csi/external-snapshotter to v3.0.3

### PR DESCRIPTION
**Pull request by tareqhs**
**How to categorize this PR?**
/area storage
/kind enhancement
/platform gcp

**What this PR does / why we need it**:
Upgrade csi snapshotter (external-snapshotter controller) to v3 to acquire bug fixes and improvements.

**Which issue(s) this PR fixes**:
Fixes https://github.com/kubernetes-csi/external-snapshotter/issues/349
The fix is available in v3 of external-snapshotter controller.

**Release note**:
```feature user
The following images are updated (see [CHANGELOG](https://github.com/kubernetes-csi/external-snapshotter/blob/v3.0.3/CHANGELOG/CHANGELOG-3.0.md) for more details):
- k8s.gcr.io/sig-storage/csi-snapshotter: v2.1.5 -> v3.0.3
- k8s.gcr.io/sig-storage/snapshot-controller: v2.1.5 -> v3.0.3
```

### CODE REVIEW

**Review by dkistner — APPROVED** [review]
/lgtm
/invite @ialidzhikov

**Comment by tareqhs** [comment]
hi @dkistner @ialidzhikov any update on this?

**Comment by prashanth26** [comment]
HI @tareqhs ,

Apologies for the delay. I think @ialidzhikov is on a vacation and should be back in the next few days. Waiting for his approval.

**Review by ialidzhikov — APPROVED** [review]
/lgtm

**Pull request merged**
